pkg/detectors/autoklose/autoklose.go

```diff
@@ -63,7 +66,15 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				defer res.Body.Close()
 				if res.StatusCode >= 200 && res.StatusCode < 300 {
 					if json.Valid(bodyBytes) {
-						s1.Verified = true
+						var responseBody map[string]interface{}
+						if err := json.Unmarshal(bodyBytes, &responseBody); err == nil {
+							s1.Verified = true
```

Comment:
Result should be marked as verified as soon as we get expected status code. Extra Data related things should be handled afterwards. In this case if for some reason un-marshal fails the result will be marked as unverified even though we have received a 200.

Reply:
Hmm, that seems right. I observed that API returns 200 but HTML response if the token is expired, so I have set that condition that if the code is not able to parse the response as JSON, it is probably invalid.

Reply:
Since we’ve added the content-type header as application/json, the response will be in JSON format. The logic should be: if we receive a 200 status code, we should unmarshal the response and capture the email. If we receive a 401, it indicates the token is expired or invalid, and we should mark it as unverified. Any status code other than 200 or 401 should be handled separately.

Reply:
nope, even after setting `Content-Type: application/json` we are getting HTML response, the server doesn't respect client headers.

<img width="879" alt="Screenshot 2024-10-17 at 4 32 05 PM" src="https://github.com/user-attachments/assets/fe0c182b-1397-4aea-8ab5-eaf237acdd92">

Reply:
We need to set `Accept` header as `application/json` in order to tell server we are expecting a JSON response.

Reply:
Can you check now if it is correct @kashifkhan0771 ?